src/components/NewCommentForm.tsx

```
@@ -1,8 +1,83 @@
-import React from 'react';
+import { useState } from 'react';
+import classNames from 'classnames';
+import { addComment } from '../api/request';
+import { Comment } from '../types/Comment';
+
+type NewCommentFormProps = {
+  selectedPostId: number | undefined;
+  onComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export function NewCommentForm({
+  selectedPostId,
+  onComments,
+}: NewCommentFormProps) {
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  function validateForm(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    const hasNameError = name.trim().length === 0;
+    const hasEmailError = email.trim().length === 0;
+    const hasBodyError = body.trim().length === 0;
+
+    setNameError(hasNameError);
+    setEmailError(hasEmailError);
+    setBodyError(hasBodyError);
+
+    const hasErrors = hasNameError || hasEmailError || hasBodyError;
+
+    if (hasErrors) {
+      return;
+    }
+
+    if (selectedPostId) {
+      const comment = {
+        postId: selectedPostId,
+        name: name,
+        email: email,
+        body: body,
+      };
+
+      setIsSubmitting(true);
+
+      addComment(comment)
+        .then((response: Comment) => {
+          const newComment: Comment = {
+            id: response.id,
+            postId: response.postId,
+            name: response.name,
+            email: response.email,
+            body: response.body,
+          };
+
+          onComments(comments => [...comments, newComment]);
+          setBody('');
+        })
+        .catch(() => {})
```

The `catch` block is currently empty, which means any errors during the `addComment` request will be silently ignored. Consider adding error handling logic here to inform the user or log the error.